lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -682,22 +684,57 @@ def local_user
       ENV['USER']
     end
 
-    # Returns the KeyPair for this host, creating it if needed
+    # Creates the KeyPair for this test run
     #
     # @param region [AWS::EC2::Region] region to create the key pair in
     # @return [AWS::EC2::KeyPair] created key_pair
     # @api private
     def ensure_key_pair(region)
-      @logger.notify("aws-sdk: Ensure key pair exists, create if not")
       key_pairs = region.key_pairs
       pair_name = key_name()
+      delete_key_pair(key_pairs, pair_name, region.name)
+      create_new_key_pair(key_pairs, pair_name)
+    end
+
+    # Deletes key pairs from all regions
+    #
+    # @return nil
+    # @api private
+    def delete_key_pair_all_regions
+      pair_name = key_name()
+      @ec2.regions.each do |region|
+        delete_key_pair(region.key_pairs, pair_name, region.name)
+      end
+    end
+
+    # Deletes a given key pair
+    #
+    # @param [Hash{String=>AWS::EC2::KeyPairCollection}] key_pairs the key pairs of a
+    #   given region, which the deleted key will be selected from
+    # @param [String] pair_name the name of the key to be deleted
+    # @param [String] region_name the name of the region to delete from
+    #
+    # @api private
+    def delete_key_pair(key_pairs, pair_name, region_name)
```

Comment:
Given `key_pairs` and `name` both always come from `region` this could be simplified to just passing the region and the pair_name. This would simplify the code and remove the duplication on lines 704, 706, 693 and 695.